File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-214358-38088/analysis/own-tests-project/src/main/java/surveycrawler/RetreatCommand.java
package surveycrawler;

public class RetreatCommand extends RepeatableCrawlerCommand {

    // testing

    @Override
    public boolean isFor(char aCommand) {
        return aCommand == 't';
    }

    // executing

    @Override
    protected void executeOnce(SurveyCrawler aCrawler) {
        aCrawler.retreat();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-213927-34997/analysis/acceptance-project/src/surveycrawler/retreat_command.py
from surveycrawler.crawler_command import CrawlerCommand


class RetreatCommand(CrawlerCommand):

    # command character

    @classmethod
    def character(cls):
        return 't'

    # executing

    def execute(self, a_crawler):
        a_crawler.retreat()

    def undo(self, a_crawler):
        a_crawler.undo_retreat()
